src/Table/Table.tsx

```diff
@@ -0,0 +1,977 @@
+                        props.onActionDropped(e);
+                    }}
+                >
+                    <div className="icon-holder" id={"ellipsis-" + props.selectedRow.rowIndex}>
+                        {ellipsis}
+                    </div>
+                    <div
+                        className={getActionColumnClasses()}
+                        ref={actionRef}
+                    >
+                        {props.actionLinks.map((link: ActionLinkItem, index: number) =>
+                            <a
+                                href="#"
+                                key={`${link.label.replace(" ", "-")} - ${index}}`}
+                                onClick={(e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+                                    e.preventDefault();
+                                    link.onClick(e, props.selectedRow);
+                                }}
+                            >
+                                {link.label}
+                            </a>
+                        )}
+                    </div>
+                </div>
+            }
+        </div>
+    );
+};
+
+interface TableUIProps {
+    actionLinks?: Array<ActionLinkItem>;
+    allRowsAreSelected?: boolean;
+    className: string;
+    columns: Array<TableHeader>;
+    footer: React.ReactNode;
+    loading: boolean;
+    onActionDropped: (event: React.MouseEvent<HTMLDivElement, MouseEvent>, row: TableRow, rowIndex?: number) => void;
+    onAllItemsSelected?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+    onItemSelected?: (e: React.ChangeEvent<HTMLInputElement>, row: TableRow, type: "row" | "subRow", rowIndex?: number) => void;
+    onRowExpanded?: (e: React.MouseEvent<HTMLDivElement, MouseEvent>, row: TableRow) => void;
+    onSort?: (accessor: string, sortDirection: sortDirectionTypes) => void;
+    onSubRowExpanded?: (e: React.MouseEvent<HTMLDivElement, MouseEvent>, row: TableRow, rowIndex: number) => void;
+    primaryActionButton?: PrimaryActionButton;
+    rows: Array<TableRow>;
+    rowsAreCollapsable?: boolean;
+    sortable: boolean;
+    useRowCollapse: boolean;
+    useRowSelection: boolean;
+    useShowActionColumn: boolean;
+}
+
+const TableUI: React.FunctionComponent<TableUIProps> = React.memo((props: TableUIProps): React.ReactElement<void> => {
+    const checkAllRandomIds: string = randomId("chk-all");
+    const tableRef: React.RefObject<HTMLTableElement> = React.createRef<HTMLTableElement>();
+    return (
+        <div className={"table-responsive" + (props.loading ? " skeleton-loader skeleton-loader-table" : "")}>
+            <table className={"table" + (props.className ? ` ${props.className}` : "")} ref={tableRef}>
+                <thead>
+                    <tr>
+                        {props.useRowSelection ?
+                            <th>
+                                <div className="custom-control custom-checkbox">
+                                    <input
```

Comment:
Would be possible to use CheckBox component here ?

Reply:
Difference component @omarVodiaK . you can't mix the two.